src/o365/spo/commands/site/site-add.ts

```diff
@@ -192,61 +217,342 @@ class SpoSiteAddCommand extends SpoCommand {
+          if (typeof error === 'boolean') {
+            resolve(error);
+          }
+          else {
+            reject(error);
+          }
+        });
+    });
+  }
+
+  private deleteSiteFromTheRecycleBin(url: string, wait: boolean, cmd: CommandInstance): Promise<void> {
+    return new Promise<void>((resolve: () => void, reject: (error: any) => void): void => {
+      this
+        .ensureFormDigest(this.spoAdminUrl as string, cmd, this.context, this.debug)
+        .then((res: FormDigestInfo): Promise<string> => {
+          this.context = res;
+
+          if (this.verbose) {
+            cmd.log(`Deleting site ${url} from the recycle bin...`);
+          }
+
+          const requestOptions: any = {
+            url: `${this.spoAdminUrl as string}/_vti_bin/client.svc/ProcessQuery`,
+            headers: {
+              'X-RequestDigest': this.context.FormDigestValue
+            },
+            body: `<Request AddExpandoFieldTypeSuffix="true" SchemaVersion="15.0.0.0" LibraryVersion="16.0.0.0" ApplicationName="${config.applicationName}" xmlns="http://schemas.microsoft.com/sharepoint/clientquery/2009"><Actions><ObjectPath Id="185" ObjectPathId="184" /><Query Id="186" ObjectPathId="184"><Query SelectAllProperties="false"><Properties><Property Name="IsComplete" ScalarProperty="true" /><Property Name="PollingInterval" ScalarProperty="true" /></Properties></Query></Query></Actions><ObjectPaths><Method Id="184" ParentId="175" Name="RemoveDeletedSite"><Parameters><Parameter Type="String">${Utils.escapeXml(url)}</Parameter></Parameters></Method><Constructor Id="175" TypeId="{268004ae-ef6b-4e9b-8425-127220d84719}" /></ObjectPaths></Request>`
+          };
+
+          return request.post(requestOptions);
+        })
+        .then((res: string): void => {
+          const json: ClientSvcResponse = JSON.parse(res);
+          const response: ClientSvcResponseContents = json[0];
+          if (response.ErrorInfo) {
+            reject(response.ErrorInfo.ErrorMessage);
+          }
+          else {
+            const operation: SpoOperation = json[json.length - 1];
+            let isComplete: boolean = operation.IsComplete;
+            if (!wait || isComplete) {
+              resolve();
+              return;
+            }
+
+            setTimeout(() => {
+              this.waitUntilFinished(JSON.stringify(operation._ObjectIdentity_), this.spoAdminUrl as string, resolve, reject, cmd, this.context as FormDigestInfo, this.dots, this.timeout);
+            }, operation.PollingInterval);
+          }
+        });
+    });
+  }
+
   public options(): CommandOption[] {
     const options: CommandOption[] = [
       {
         option: '--type [type]',
-        description: 'Type of modern sites to list. Allowed values TeamSite|CommunicationSite, default TeamSite',
-        autocomplete: ['TeamSite', 'CommunicationSite']
+        description: 'Type of modern sites to list. Allowed values TeamSite|CommunicationSite|ClassicSite, default TeamSite',
```

Comment:
This description is incorrect. It should be something along the lines of: _Type of site to create..._